var/spack/repos/builtin/packages/py-oauthlib/package.py

```diff
@@ -15,6 +15,7 @@ class PyOauthlib(PythonPackage):
     homepage = "https://github.com/oauthlib/oauthlib"
     pypi = "oauthlib/oauthlib-3.1.0.tar.gz"
 
+    version("3.2.1", sha256="1565237372795bf6ee3e5aba5e2a85bd5a65d0e2aa5c628b9a97b7d7a0da3721")
     version("3.1.1", sha256="8f0215fcc533dd8dd1bee6f4c412d4f0cd7297307d43ac61666389e3bc3198a3")
     version("3.1.0", sha256="bee41cc35fcca6e988463cacc3bcb8a96224f470ca547e697b604cc697b2f889")
     version("3.0.1", sha256="0ce32c5d989a1827e3f1148f98b9085ed2370fc939bf524c9c851d8714797298")
```

Comment:
cryptography no longer has an upper bound. Honestly, we should prob split `+extras` into 3 separate variants and make them off by default to match the behavior of pip.